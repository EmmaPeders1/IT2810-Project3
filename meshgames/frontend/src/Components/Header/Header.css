
.header {
  background: linear-gradient(to bottom, #9999ff 0%, #ff3399 100%);
  height: 22vh;
  width: 100vw;
  font-size: 40px;
  font-family: 'Courier New', Courier, monospace;
  color: white;
  position: sticky;
  display: flex;
  flex-flow: column wrap;
  top: 0;
  margin: 0;
  overflow: hidden;
  box-shadow: 3px 3px 10px 6px whitesmoke;
  justify-content: center;
}

h1{
  padding: 1rem;
}

/* html, body {
    overflow-x: hidden;
    height: 100%;
} */
body {
    background: #fff;
    padding: 0;
    margin: 0;
    font-family: 'Courier New', Courier, monospace
}

.main {
    margin: 0 auto;
    display: block;
    height: 100%;
    margin-top: 60px;
}
.mainInner{
    display: table;
    height: 100%;
    width: 100%;
    text-align: center;
}
.mainInner div{
    display:table-cell;
    vertical-align: middle;
    font-size: 3em;
    font-weight: bold;
    letter-spacing: 1.25px;
}
#sidebarMenu {
    height: 100vh;
    position: fixed;
    left: 0;
    width: 250px;
    margin-top: 490px;
    display: flex;
    transform: translateX(-250px);
    transition: transform 250ms ease-in-out;
    background: linear-gradient(to bottom, #ff3399 0%, #9999ff 100%);
}
.sidebarMenuInner{
    margin:75;
    padding:0;
    border-top: 1px solid rgba(255, 255, 255, 0.10);
}
.sidebarMenuInner li{
    list-style: none;
    color: #fff;
    text-transform: uppercase;
    font-weight: bold;
    padding: 20px;
    cursor: pointer;
    border-bottom: 1px solid rgba(255, 255, 255, 0.10);
}
.sidebarMenuInner li span{
    display: block;
    font-size: 14px;
    color: rgba(255, 255, 255, 0.50);
}
.sidebarMenuInner li a{
    color: #fff;
    text-transform: uppercase;
    font-weight: bold;
    cursor: pointer;
    text-decoration: none;
}
input[type="checkbox"]:checked ~ #sidebarMenu {
    transform: translateX(0);
}

input[type=checkbox] {
    transition: all 0.3s;
    box-sizing: border-box;
    display: none;
}
.sidebarIconToggle {
    transition: all 0.3s;
    box-sizing: border-box;
    cursor: pointer;
    position: absolute;
    z-index: 99;
    height: 100%;
    width: 100%;
    top: 22px;
    left: 15px;
    height: 22px;
    width: 22px;
}
.spinner {
    transition: all 0.3s;
    box-sizing: border-box;
    position: absolute;
    height: 3px;
    width: 100%;
    background-color: #fff;
}
.horizontal {
    transition: all 0.3s;
    box-sizing: border-box;
    position: relative;
    float: left;
    margin-top: 3px;
}
.diagonal.part-1 {
    position: relative;
    transition: all 0.3s;
    box-sizing: border-box;
    float: left;
}
.diagonal.part-2 {
    transition: all 0.3s;
    box-sizing: border-box;
    position: relative;
    float: left;
    margin-top: 3px;
}
input[type=checkbox]:checked ~ .sidebarIconToggle > .horizontal {
    transition: all 0.3s;
    box-sizing: border-box;
    opacity: 0;
}
input[type=checkbox]:checked ~ .sidebarIconToggle > .diagonal.part-1 {
    transition: all 0.3s;
    box-sizing: border-box;
    transform: rotate(135deg);
    margin-top: 8px;
}
input[type=checkbox]:checked ~ .sidebarIconToggle > .diagonal.part-2 {
    transition: all 0.3s;
    box-sizing: border-box;
    transform: rotate(-135deg);
    margin-top: -9px;
}