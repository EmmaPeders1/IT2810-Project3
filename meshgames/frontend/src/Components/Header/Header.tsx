import React from 'react';
import './Header.css';

function Header() {
  return (
    <div className="header">
       <h1>MeshGames</h1>
    </div>
  );
}

export default Header;