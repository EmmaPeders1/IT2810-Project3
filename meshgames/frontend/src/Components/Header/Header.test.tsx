import { render } from "@testing-library/react";
import React from 'react';
import Header from './Header';

test('Check if header is not empty', () => {
  const {container} = render(<Header />);
  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("MeshGames");
})
