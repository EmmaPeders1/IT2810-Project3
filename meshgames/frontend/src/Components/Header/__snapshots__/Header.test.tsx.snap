// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if header is not empty 1`] = `
<div>
  <div
    class="header"
  >
    <h1
      id="headerText"
    >
      MeshGames
    </h1>
  </div>
</div>
`;
